app/jobs/shipit/github_sync_job.rb

```diff
@@ -32,7 +32,8 @@ def append_commit(gh_commit)
 
     def fetch_missing_commits(&block)
       commits = []
-      iterator = Shipit::FirstParentCommitsIterator.new(&block)
+      github_api = @stack&.repository&.github_api || Shipit.github.api # TODO: figure out something cleaner
```

Comment:
Is the `||` necessary here? Looks like `@stack` can never be `nil`. Stacks should also have a repository associated with them in all cases.

Reply:
This is purely for making a particular test pass. We've done something similar in other parts of the code (search for the comment).